content/academy/puppeteer_playwright/reading_intercepting_requests.md

```diff
@@ -0,0 +1,308 @@
+});
+
+await page.goto('https://soundcloud.com/tiesto/following');
+
+await page.waitForTimeout(10000);
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+const page = await browser.newPage();
+
+// Listen for all requests
+page.on('request', (req) => {
+    // If the URL doesn't include our keyword, ignore it
+    if (!req.url().includes('followings')) return;
+
+    // Convert the request URL into a URL object
+    const url = new URL(req.url());
+
+    // Print the search parameters in object form
+    console.log(Object.fromEntries(url.searchParams));
+});
+
+await page.goto('https://soundcloud.com/tiesto/following');
+
+await page.waitForTimeout(10000);
+await browser.close();
+</marked-tab>
+```
+
+After running this code, we can see this logged to the console:
+
+```text
+{
+  client_id: 'llCGDUjKpxUslgO1yEce7Zh95PXE78Bo',
+  limit: '12',
+  offset: '0',
+  linked_partitioning: '1',
+  app_version: '1652347025',
+  app_locale: 'en'
+}
+```
+
+## [](#reading-responses) Reading responses
+
+Listening for and reading responses is very similar to reading requests. The only difference is that we need to listen for the **response** event instead of **request**. Additionally, the object passed into the callback function represents the response instead of the request.
+
+<!-- To access the response object from **Route** in Playwright, we need to wait for it with `await route.request().response()`. Nothing else needs to change. -->
+
+This time, instead of grabbing the query parameters of the request URL, let's grab hold of the response body and print it to the console in JSON format:
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+// Notice that the callback function is now async
+page.on('response', async (res) => {
+    if (!res.request().url().includes('followings')) return;
+
+    // Grab the response body in JSON format
+    const json = await res.json();
+    console.log(json);
+});
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+// Notice that the callback function is now async
+page.on('response', async (res) => {
+    if (!res.request().url().includes('followings')) return;
+
+    // Grab the response body in JSON format
+    const json = await res.json();
+    console.log(json);
+});
+</marked-tab>
+```
+
+Upon running this code, we'll see the API response logged into the console:
+
+![API response in console]({{@asset puppeteer_playwright/images/api-response-tiesto.webp}})
+
+## [](#intercepting-requests) Intercepting requests
+
+One of the most popular ways of speeding up website loading in Puppeteer and Playwright is by blocking certain resources from loading. These resources are usually CSS files, images, and other miscellanous resources that aren't super necessary (mainly because the computer doesn't have eyes - it doesn't care how the website looks!).
```

Comment:
Im totally missing the part about cache being turned off by turning on interception and how that's dangerous.

For blocking requests, there's [a completely different way of doing that that we have in the SDK](https://github.com/apify/apify-js/blob/55e0cce99dc7bfc5c303bb76885cdaf037ff6cd7/src/puppeteer_utils.js#L187) which does not turn off cache. It would be nice to mention the CDP method here as the correct way for blocking requests when performance is important. Also pls mention that it only works in Chromium based browsers.